helm/resource_release.go

```diff
@@ -694,107 +694,93 @@ func resourceDiff(ctx context.Context, d *schema.ResourceDiff, meta interface{})
-	client.ResetValues = d.Get("reset_values").(bool)
-	client.ReuseValues = d.Get("reuse_values").(bool)
-	client.Recreate = d.Get("recreate_pods").(bool)
-	client.MaxHistory = d.Get("max_history").(int)
-	client.CleanupOnFail = d.Get("cleanup_on_fail").(bool)
-	client.Description = d.Get("description").(string)
-
-	if cmd := d.Get("postrender.0.binary_path").(string); cmd != "" {
-		pr, err := postrender.NewExec(cmd)
 
-		if err != nil {
-			return err
+		debug("%s performing dry run", logID)
+
+		client := action.NewUpgrade(actionConfig)
+		client.ChartPathOptions = *cpo
+		client.Devel = d.Get("devel").(bool)
+		client.Namespace = d.Get("namespace").(string)
+		client.Timeout = time.Duration(d.Get("timeout").(int)) * time.Second
+		client.Wait = d.Get("wait").(bool)
+		client.DryRun = true // do not apply changes
+		client.DisableHooks = d.Get("disable_webhooks").(bool)
+		client.Atomic = d.Get("atomic").(bool)
+		client.SubNotes = d.Get("render_subchart_notes").(bool)
+		client.Force = d.Get("force_update").(bool)
+		client.ResetValues = d.Get("reset_values").(bool)
+		client.ReuseValues = d.Get("reuse_values").(bool)
+		client.Recreate = d.Get("recreate_pods").(bool)
+		client.MaxHistory = d.Get("max_history").(int)
+		client.CleanupOnFail = d.Get("cleanup_on_fail").(bool)
+		client.Description = d.Get("description").(string)
+
+		if cmd := d.Get("postrender.0.binary_path").(string); cmd != "" {
+			pr, err := postrender.NewExec(cmd)
+			if err != nil {
+				return err
+			}
+			client.PostRenderer = pr
 		}
 
-		client.PostRenderer = pr
-	}
-
-	values, err := getValues(d)
-	if err != nil {
-		return fmt.Errorf("error getting values for a diff: %v", err)
-	}
-
-	dry, err := client.Run(name, chart, values)
-	if err != nil && strings.Contains(err.Error(), "has no deployed releases") {
-		if len(chart.Metadata.Version) > 0 {
-			return d.SetNew("version", chart.Metadata.Version)
+		values, err := getValues(d)
+		if err != nil {
+			return fmt.Errorf("error getting values for a diff: %v", err)
 		}
-		d.SetNewComputed("version")
 
-		if m.ExperimentEnabled("manifest") {
+		dry, err := client.Run(name, chart, values)
+		if err != nil && strings.Contains(err.Error(), "has no deployed releases") {
+			if len(chart.Metadata.Version) > 0 {
+				return d.SetNew("version", chart.Metadata.Version)
+			}
+			d.SetNewComputed("version")
 			d.SetNewComputed("manifest")
-		} else {
-			d.Clear("manifest")
+			return nil
+		} else if err != nil {
+			return fmt.Errorf("error running dry run for a diff: %v", err)
 		}
 
-		return nil
-	} else if err != nil {
-		return fmt.Errorf("error running dry run for a diff: %v", err)
-	}
-
-	if m.ExperimentEnabled("manifest") {
 		jsonManifest, err := convertYAMLManifestToJSON(dry.Manifest)
 		if err != nil {
 			return err
 		}
 		manifest := redactSensitiveValues(string(jsonManifest), d)
 		d.SetNew("manifest", manifest)
 		debug("%s set manifest: %s", logID, jsonManifest)
+	} else {
```

Comment:
Some would say it's considered idiomatic in Go to minimise the level of indentations and the size of the sub-branches. I think if the logic is inverted in this conditional and a return added after d.Clear("manifest") it should let the rest of the code live on the same level as it was before. Just an opinion, though.

Reply:
I agree with this idiom in general, but as this is a feature toggle not business logic I feel like there is value having the code path behind the toggle be nested in an obvious way like this.